### PR TITLE
feat: admin can delete single user account

### PR DESCRIPTION
**Pull request by PiTrem**
image

refactor!: user managment
- better notification (no alert)
image

 - clean api endpoints:
 ```
                                Find all users     GET  /api/v1/admin/users 
 Find top (5) matched user by name and by type     GET  /api/v1/admin/users/byname 
                  restore deleted user account    POST  /api/v1/admin/users/restoreAccount 
                       create new user account    POST  /api/v1/admin/users 
                               Find user by ID     GET  /api/v1/admin/users/:user_id 
                           update user account     PUT  /api/v1/admin/users/:user_id 
                           delete user account  DELETE  /api/v1/admin/users/:user_id 
                           reset user password     PUT  /api/v1/admin/users/:user_id/resetPassword
                              get user profile     GET  /api/v1/admin/users/:user_id/profile
 ```

### CODE REVIEW

File: app/packs/src/apps/admin/UserManagement.js [hunk 1]
```diff
@@ -413,17 +420,46 @@ export default class UserManagement extends React.Component {
...
+        }
+        // remove deleted user from this.state.users
+        const { users } = this.state;
+        const index = users.findIndex((u) => u.id === user.id);
+        if (index > -1) {
+          users.splice(index, 1);
+        }
+        this.setState({
+          showEditUserModal: false,
+          alertMessage: `User with name abbreviation ${user.nameAbbreviation} has been deleted!`,
```
Comment (f-idiris) [inline app/packs/src/apps/admin/UserManagement.js:455]: user.nameAbbreviation returns undefined (see image below). 
abrev_undefined

The abbreviation is actually stored in **user.initials.**

**Review by f-idiris — CHANGES_REQUESTED** [review]

**Review by f-idiris — COMMENTED** [review]
When you delete an account and then restore it, the elements are not fetched and you get a blank page.  I am not sure if this is due to this PR or if we need to open a separate issue.
create_account_delete_then_restore

File: app/packs/src/apps/admin/UserManagement.js (hunk 1 above)
Comment (PiTrem) [reply app/packs/src/apps/admin/UserManagement.js]: corrected :heavy_check_mark:

**Comment by PiTrem** [comment]
@f-idiris 
> When you delete an account and then restore it, the elements are not fetched and you get a blank page. I am not sure if this is due to this PR or if we need to open a separate issue. 

resolved with  recreating user profile
https://github.com/ComPlat/chemotion_ELN/pull/1883/commits/26e305e9baaafc3c4cb27b150404245568202547

**Review by f-idiris — APPROVED** [review]

**Pull request merged**